From 34d8e6bf37155488c61029cf392255dd18ed0a87 Mon Sep 17 00:00:00 2001
From: Huisong Li <lihuisong@huawei.com>
Date: Fri, 21 Oct 2022 15:36:54 +0800
Subject: [PATCH 178/189] app/testpmd: fix RSS types display

Now testpmd fails to display types when query RSS rule. The failure is
because the '\n' character is missing at the end of the function
'rss_config_display()'.
Actually, all places calling 'xxx_types_display()' need to '\n'. So this
patch moves '\n' to the inside of these function.

Bugzilla ID: 1048
Fixes: 534988c490f1 ("app/testpmd: unify RSS types display")
Fixes: 44a37f3cffe0 ("app/testpmd: compact RSS types output")

Signed-off-by: Huisong Li <lihuisong@huawei.com>
Tested-by: Weiyuan Li <weiyuanx.li@intel.com>
Reviewed-by: Ferruh Yigit <ferruh.yigit@xilinx.com>
---
 app/test-pmd/config.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/app/test-pmd/config.c b/app/test-pmd/config.c
index f8cd135970..12386c4d82 100644
--- a/app/test-pmd/config.c
+++ b/app/test-pmd/config.c
@@ -731,6 +731,7 @@ rss_offload_types_display(uint64_t offload_types, uint16_t char_num_per_line)
 			total_len += str_len;
 		}
 	}
+	printf("\n");
 }
 
 void
@@ -840,7 +841,6 @@ port_infos_display(portid_t port_id)
 		printf("Supported RSS offload flow types:\n");
 		rss_offload_types_display(dev_info.flow_type_rss_offloads,
 				TESTPMD_RSS_TYPES_CHAR_NUM_PER_LINE);
-		printf("\n");
 	}
 
 	printf("Minimum size of RX buffer: %u\n", dev_info.min_rx_bufsize);
@@ -1604,6 +1604,7 @@ rss_types_display(uint64_t rss_types, uint16_t char_num_per_line)
 			total_len += str_len;
 		}
 	}
+	printf("\n");
 }
 
 static void
@@ -3102,7 +3103,6 @@ port_rss_hash_conf_show(portid_t port_id, int show_rss_key)
 	}
 	printf("RSS functions:\n");
 	rss_types_display(rss_hf, TESTPMD_RSS_TYPES_CHAR_NUM_PER_LINE);
-	printf("\n");
 	if (!show_rss_key)
 		return;
 	printf("RSS key:\n");
-- 
2.23.0

